broker-daemon/index.js

```diff
@@ -183,20 +183,16 @@ class BrokerDaemon {
    */
   async initialize () {
     try {
-      // Since these are potentially long-running operations, we run them in parallel to speed
-      // up BrokerDaemon startup time.
-      await Promise.all([
-        this.initializeMarkets(this.marketNames),
-        (async () => {
-          this.logger.info(`Initializing BlockOrderWorker`)
-          await this.blockOrderWorker.initialize()
-          this.logger.info('BlockOrderWorker initialized')
-        })()
-      ])
+      await this.initializeMarkets(this.marketNames)
 
-      // This will run in the background. It is implemented with exponential backoff so
-      // the validation will be retried on the engine until successful or until final failure.
-      this.validateEngines()
+      // Starts the validation of all engines on the broker. We do not await this
+      // function because we want the validations to run in the background as it
+      // can take time for the engines to be ready
+      const enginesAreValidated = this.validateEngines()
+
+      // We run this asynchronously so that it doesn't block the start of the rpc
+      // server. This function will not return until the engines are validated
+      this.initializeBlockOrder(enginesAreValidated)
```

Comment:
Any way we can await only the first part of this? What happens if the RPC starts but the block orders are not yet indexed?